File: Python3/Best Time to Buy and Sell Stock.py
# 44ms, 100.0%
class Solution:
    def maxProfit(self, prices):
        max_profit = beg = 0
        for i in range(1, len(prices)):
            if prices[i] <= prices[beg]:
                beg = i
            else:
                max_profit = max(max_profit, prices[i] - prices[beg])
        return max_profit
        

File: Java/Best Time to Buy and Sell Stock.java
// 1ms, 99.26%; 39.4MB, 22.56%
class Solution {
    public static int maxProfit(int[] prices) {
        int res = 0, min = Integer.MAX_VALUE;
        for (int val : prices) {
            if (val < min) {
                min = val;
            } else {
                res = Math.max(res, val - min);
            }
        }
        return res;
    }
}
